### ISSUE TITLE
Webpack or Vuex seems to break the build

### ISSUE THREAD

**Issue by JLKM** [yyss8/Scms#2]
After correcting a typo in `components\header\headersetting.vue` `</buttuon>` and some missing v-bind:key directives in `components/header.vue` I managed to get through a `npm run dev`. But with no result.

Message on `localhost:8085`:

```
Vue.js error

TypeError: Cannot read property 'headerLink' of null
    at loadSetting (__vue_ssr_bundle__:2966:39)
    at wrappedMutationHandler (/var/www/node/scblog/node_modules/vuex/dist/vuex.js:597:5)
    at commitIterator (/var/www/node/scblog/node_modules/vuex/dist/vuex.js:321:7)
    at Array.forEach (<anonymous>)
    at /var/www/node/scblog/node_modules/vuex/dist/vuex.js:320:11
    at Store._withCommit (/var/www/node/scblog/node_modules/vuex/dist/vuex.js:408:3)
    at Store.commit (/var/www/node/scblog/node_modules/vuex/dist/vuex.js:319:8)
    at boundCommit (/var/www/node/scblog/node_modules/vuex/dist/vuex.js:275:19)
    at nuxtServerInit (__vue_ssr_bundle__:3016:13)
    at Array.wrappedActionHandler (/var/www/node/scblog/node_modules/vuex/dist/vuex.js:604:15)
```

Exactly the same screen-message on my Ubuntu and Windows setup (both with Node 8.X). A little difference in the console, though.

On Windows 10 the systems goes stuck after this:
```
Server Started :: 8085
(node:8344) DeprecationWarning: loaderUtils.parseQuery() received a non-string value which can be problematic, see https://github.com/webpack/loader-utils/issues/56
parseQuery() will be replaced with getOptions() in the next major version of loader-utils.
webpack built 1d5e085f536fabeb2c92 in 11515ms
webpack building...
webpack built 1d5e085f536fabeb2c92 in 822ms
```

On Ubuntu 16.04 LTS the same depreciation warning about webpack but then a dozen attempts (resultless it seems) to deliver the page:

```
root@jlkm:/var/www/node/scblog# npm run dev

> scms@1.0.0 dev /var/www/node/scblog
> node app.js

Server Started :: 8085
  build [==                  ] 10%(node:7693) DeprecationWarning: loaderUtils.parseQuery() received a non-string value which can be problematic, see https://github.com/webpack/loader-utils/issues/56
parseQuery() will be replaced with getOptions() in the next major version of loader-utils.
Build completed in 5.804s

webpack built 99e0c0a39167d6002031 in 5808ms
webpack building...

Build completed in 0.177s

webpack built 99e0c0a39167d6002031 in 178ms
webpack building...

Build completed in 0.152s

webpack built 99e0c0a39167d6002031 in 152ms
webpack building...

Build completed in 0.149s

webpack built 99e0c0a39167d6002031 in 150ms
webpack building...

Build completed in 0.149s

webpack built 99e0c0a39167d6002031 in 149ms
webpack building...

Build completed in 0.135s

webpack built 99e0c0a39167d6002031 in 135ms
webpack building...

Build completed in 0.158s

webpack built 99e0c0a39167d6002031 in 159ms
webpack building...

Build completed in 0.139s

webpack built 99e0c0a39167d6002031 in 140ms
webpack building...

Build completed in 0.133s

webpack built 99e0c0a39167d6002031 in 133ms
webpack building...

Build completed in 0.143s

webpack built 99e0c0a39167d6002031 in 144ms
webpack building...

Build completed in 0.139s

webpack built 99e0c0a39167d6002031 in 140ms
webpack building...

Build completed in 0.146s

webpack built 99e0c0a39167d6002031 in 146ms
webpack building...

Build completed in 0.129s

webpack built 99e0c0a39167d6002031 in 129ms
```

By combining MongoDB, Express, Nuxt and SSR your repo looks like a real good buildingstone for a Vue-CMS. However it looks like, just a minor problem is preventing us from getting there.

Hopefully this feedback will push your project a little further.

Please return, if I can be of more help.

**Comment by yyss8** [comment]
Hi, JLKM, thank you for letting me know those issues.
I'm surprised that someone would even pay attention to this simple project that I wrote while learning the Vue SSR. I haven't updated anything for project and even my github since I found a job as a php developer

Anyway, for the first error that you found, I've uploaded a mongodb database sample,  just simply insert it into your database and change the database config file in config/db.config should fix the error.

and for the warning, I think it's because of using out-dated libraries, most likely the Nuxt, because it has integrated the loaders, vue-ssr and webpack e.g... try to do a npm update to see if it could fix it . However, I don't think it would block the webpack from building because it's just a depreciation warning.

Unfortunately, I didn't set up the multiple language plugin at that time, I'm afraid that you might not be able to understand most of the server response messages and comments that I wrote but I think it shouldn't be too hard to guess the purpose of those functions by reading its name.

PS: try to run it by using command "node app", 

Hope this helps

**Comment by JLKM** [comment]
Much appreciate your fast and careful help.

In the first place I moved on by building on the newest version of Nuxt, 1.0.0.-rc6, but it went all haywire with lost dependencies in especially webpack.

Then I locked the setup on Nuxt 0.9.6 like you, imported your data into MongoDB, some more setup stuff...

...and Bingo! Your 3 page blog with chinese letters is up and running.

Now I'm going to dig further into categories, design and other things. And who knows, how long it'll take to put on tags?

Yes. Your project might not be the biggest one on GitHub. But other repos in the same area are either missing MongoDB (they use markdown), using Koa (which I won't), or they fall short in server side rendering. This goes at least for the repos, I've been able to discover.

Your small project hits the sweet spot in my view, and I'm quite eager to develop this further.

Are You still sure, PHP - and not Vue with SSR CMS - is the way to go...;-)

Thanks once again. Keep up the good work.
